src/realm/set.hpp

```diff
@@ -1240,13 +1240,16 @@ inline size_t LnkSet::find_any(Mixed value) const
 {
     if (value.is_null())
         return not_found;
-    if (value.get_type() != type_Link)
-        return not_found;
-    size_t found = find(value.get<ObjKey>());
-    if (found != not_found) {
-        found = real2virtual(found);
+    if (value.get_type() == type_Link) {
```

Comment:
This is just a mater of taste, but it seems to me that these two if can be merged into one.
```
if (value.get_type() == type_Link || 
(value.get_type() == type_TypedLink && link.get_table_key() == get_target_table()->get_key()))
{
     return find_first(value.get<ObjKey>());
}
```

Reply:
Not quite equivalent. You are missing the 'link' variable